### DISCUSSION TITLE
Is there something to do to initialize casbin ?

### DISCUSSION THREAD

**Discussion by khyoz** [sujit-baniya/fiber-boilerplate#12] [Q&A]
Hi, first thanks for your package ! I try to test it (windows 10 pro/laragaon) and once I sign in, I only move to error pages with code 500. 

Webservice currently unavailable Error 500

An unexpected condition was encountered.
Our service team has been dispatched to bring it back online.

interface conversion: interface {} is nil, not string

So do I need to create records into the casbin_rule table ? Can you provide a simple example how to fill one entry ? Or do I need to do something else ?

Thanks for support.

**Comment by sujit-baniya** [comment]
Hi @khyoz  Thank you for raising the issue.

Can you please provide more information on replicating the issue?

Regarding your question to casbin_rule,
You don't need to create any records to make it work. Once the system is setup, it should work without any issue. For your reference I've added some endpoints, from where you can add Roles and Permissions as needed:

https://github.com/sujit-baniya/fiber-boilerplate/blob/master/rest/routes/admin.go#L26-L40

**To Create Role,**

https://github.com/sujit-baniya/fiber-boilerplate/blob/master/rest/routes/admin.go#L28

```json
{
    "role": "admin",
}
```

**To Assign Role to User,**

```json
{
    "role": "admin",
    "user_id": 1
}
```

This will create entry in casbin_rul as **g, admin, 1**

**To Create Permission for a role,**

```json
{
    "role": "admin",
    "module": "blog",
    "action": "store"
}
```

This will create entry in casbin_rul as **p, 1, blog, store**

**To Create Route Permission for a role,**

```json
{
    "role": "admin",
    "route": "/blog",
    "method": "POST"
}
```

This will create entry in casbin_rul as **p, 1, /blog,POST**

Also please make sure Redis, and Database (Mysql or Postgres) is running.

**Reply by khyoz** [reply]
thanks for your answer, here is a short video, user is existing into the database, (localhost) redis is up and running, I cannot find the issue
[video](https://storyxpress.co/video/kkqotqvlpc09k2cug)

**Comment by sujit-baniya** [comment]
Hi @khyoz Can you disable 
![image](https://user-images.githubusercontent.com/4959215/106880235-3a440b00-6704-11eb-9514-8cfba3450673.png)
and try to run the app again. You should see some error trace in terminal.

Meanwhile I am trying to figure out what might go wrong

**Comment by sujit-baniya** [comment]
Here is what I got on my local: https://storyxpress.co/video/kkqqkyt5l4pgrngv4
Is it possible to share your repo (if anything changed)?

**Reply by khyoz** [reply]
I have commented the line 
blank page -> interface conversion: interface {} is nil, not string
no message in terminal
and funny, when I go back with browser to login page, press shift-refresh page on chrome, I can see the page with header, footer, menu and Hello, x - ID: 2! with the paypal box

but when I click on Files menu, same blank page and message
-> interface conversion: interface {} is nil, not string

[funny](https://storyxpress.co/video/kkqt8azy7pfp1x6aw)

**Reply by sujit-baniya** [reply]
Ah! then it might be cache issue. I'll investigate the issue

**Reply by sujit-baniya** [reply]
I've added error stack for errors. Please pull latest changes

**Reply by khyoz** [reply]
Thanks for update, now I can see the settings right after login!
But still when I click on files, I have a 500 Error

Webservice currently unavailable Error 500
An unexpected condition was encountered.
Our service team has been dispatched to bring it back online.

interface conversion: interface {} is nil, not string 

and now I can see in the terminal

`interface conversion: interface {} is nil, not string
github.com/sujit-baniya/fiber-boilerplate/config.CustomErrorHandler
        c:/GOLANG/ia2/config/server.go:334
github.com/gofiber/fiber/v2.(*App).handler
        C:/Users/noelg/go/pkg/mod/github.com/gofiber/fiber/v2@v2.4.1/router.go:156
github.com/valyala/fasthttp.(*Server).serveConn
        C:/Users/noelg/go/pkg/mod/github.com/valyala/fasthttp@v1.19.0/server.go:2166
github.com/valyala/fasthttp.(*workerPool).workerFunc
        C:/Users/noelg/go/pkg/mod/github.com/valyala/fasthttp@v1.19.0/workerpool.go:223
github.com/valyala/fasthttp.(*workerPool).getCh.func1
        C:/Users/noelg/go/pkg/mod/github.com/valyala/fasthttp@v1.19.0/workerpool.go:195
runtime.goexit
        c:/go/src/runtime/asm_amd64.s:1374interface conversion: interface {} is nil, not stringinterface conversion: interface {} is nil, not string
github.com/sujit-baniya/fiber-boilerplate/config.CustomErrorHandler
        c:/GOLANG/ia2/config/server.go:334
github.com/gofiber/fiber/v2.(*App).handler
        C:/Users/noelg/go/pkg/mod/github.com/gofiber/fiber/v2@v2.4.1/router.go:156
github.com/valyala/fasthttp.(*Server).serveConn
        C:/Users/noelg/go/pkg/mod/github.com/valyala/fasthttp@v1.19.0/server.go:2166
github.com/valyala/fasthttp.(*workerPool).workerFunc
        C:/Users/noelg/go/pkg/mod/github.com/valyala/fasthttp@v1.19.0/workerpool.go:223
github.com/valyala/fasthttp.(*workerPool).getCh.func1
        C:/Users/noelg/go/pkg/mod/github.com/valyala/fasthttp@v1.19.0/workerpool.go:195
runtime.goexit
        c:/go/src/runtime/asm_amd64.s:1374interface conversion: interface {} is nil, not string`
        
Does that help you ?
Tks for support !

**Reply by sujit-baniya** [reply]
Found the issue. The issue was related to UserID not set in context Local. Please pull the latest version

**Reply by khyoz** [reply]
Yeah, it seems to be ok now !!!
Thank you very much ! This package contains everything I like, golang, fiber, tailwind, paypal, vuejs, redis... etc
and awesome support !
